src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs

```diff
@@ -10,61 +11,58 @@
+                              :sourceType "gcs"
+                              :type (case file-extension
+                                      "bam" "alignment"
+                                      "bed" "annotation"
+                                      "vcf" "variant")
+                              :url track-url
+                              :indexURL (when (string? @index-url) @index-url)
+                              ;; NB: we'd love to skip setting indexURL and set indexed=true instead, which allows IGV to search
+                              ;; for indexes by naming convention, similar (but more lazily-loaded than) our track selector modal.
+                              ;; However, IGV's naming convention only supports (filename + ".bai"), e.g. my.bam.bai, whereas
+                              ;; sometimes we also want (filename.replace(".bam", ".bai")), e.g. my.bai.
+                              ;; So, we have keep our own track selector with index-finder.
+                              }))
                          tracks)}))
 
 (react/defc IGVContainer
-  {:component-will-mount
-   (fn []
-     (when-not @igv-styles-loaded?
-       (load-igv-styles)))
+  {
    :render
    (fn [{:keys [this state]}]
      (let [{:keys [deps-loaded? error?]} @state]
        [:div {}
-        [:div {:ref "container"}]
+        [:div {:ref "container" :data-test-id "igv-container"}]
         (cond
           error? (style/create-server-error-message "Unable to load IGV.")
-          deps-loaded? [ScriptLoader
-                        {:key "igv"
-                         :on-error #(swap! state assoc :error? true)
-                         :on-load #(do
-                                     (swap! state assoc :igv-loaded? true)
-                                     (this :refresh))
-                         :path "https://igv.org/web/release/1.0.6/igv-1.0.6.min.js"}]
           :else [ScriptLoader
-                 {:key "igv-deps"
-                  :on-error #(swap! state assoc :error? true)
-                  :on-load #(swap! state assoc :deps-loaded? true)
-                  :path "igv-deps.bundle.js"}])]))
+                  {:key "igv"
+                   :on-error #(swap! state assoc :error? true)
+                   :on-load #(do
+                               (swap! state assoc :igv-loaded? true)
+                               (this :refresh))
+                   :path "https://igv.org/web/release/2.0.0-rc5/dist/igv.min.js"}])]))
    :component-did-update
    (fn [{:keys [props state prev-props this]}]
      (when (and (not= (:tracks props) (:tracks prev-props)) (:igv-loaded? @state))
        (this :refresh)))
    :refresh
    (fn [{:keys [props refs]}]
+     ;; .createBrowser returns a Promise that we should be using to add/remove tracks. For now, we do it brute-force:
+     ;; empty the container, then create a new IGV browser each time we change tracks.
+     (set! (.-innerHTML (@refs "container")) "")
```

Comment:
I'd consider fixing this in a future PR; I don't want to do it here, to keep changes small.